From c1321370c9af9681e0604c4d3363cf362fb48598 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 16 Sep 2024 15:56:51 +0100
Subject: [PATCH 1266/1350] drm: vc4: Fix interpolate bit for nearest neighbour
 filter

Operator precedence resulted in the wrong value being written
for nearest neighbour mode. Correct it.

Fixes: d6a3a3f10601 ("drm/vc4: Add support for per plane scaling filter selection")
Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/vc4/vc4_plane.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/gpu/drm/vc4/vc4_plane.c
+++ b/drivers/gpu/drm/vc4/vc4_plane.c
@@ -623,7 +623,7 @@ static void vc4_write_ppf(struct vc4_pla
 	phase &= SCALER_PPF_IPHASE_MASK;
 
 	vc4_dlist_write(vc4_state,
-			no_interpolate ? SCALER_PPF_NOINTERP : 0 |
+			(no_interpolate ? SCALER_PPF_NOINTERP : 0) |
 			SCALER_PPF_AGC |
 			VC4_SET_FIELD(scale, SCALER_PPF_SCALE) |
 			/*
